os/net/mac/csma/csma-output.c

```diff
@@ -388,30 +391,12 @@ tx_ok(struct packet_queue *q, struct neighbor_queue *n, int num_transmissions)
 }
 /*---------------------------------------------------------------------------*/
 static void
-packet_sent(void *ptr, int status, int num_transmissions)
+packet_sent(struct neighbor_queue *n,
+    struct packet_queue *q,
+    int status,
+    int num_transmissions)
 {
-  struct neighbor_queue *n;
-  struct packet_queue *q;
-
-  n = ptr;
-  if(n == NULL) {
-    return;
-  }
-
-  /* Find out what packet this callback refers to */
-  for(q = list_head(n->packet_queue);
-      q != NULL; q = list_item_next(q)) {
-    if(queuebuf_attr(q->buf, PACKETBUF_ATTR_MAC_SEQNO) ==
-       packetbuf_attr(PACKETBUF_ATTR_MAC_SEQNO)) {
-      break;
-    }
-  }
-
-  if(q == NULL) {
-    LOG_WARN("packet sent: seqno %u not found\n",
-           packetbuf_attr(PACKETBUF_ATTR_MAC_SEQNO));
-    return;
-  } else if(q->ptr == NULL) {
+  if(q->ptr == NULL) {
```

Comment:
maybe re-add the check that `n` and `q` are non-null? (as a defensive programming measure)

Reply:
Since this function is static, I think we can dispense with defensive programming measures here.

Reply:
Right, and the current code very clearly ensures the function will not be called with `n` nor `q` set to `NULL`.

I'd personally still add the check for future code evolution, but that's up to you.